### PR TITLE
refactor: improve generator-cli test performance

### PR DESCRIPTION
**Pull request by shojisoma**
Closes https://github.com/SAP/cloud-sdk-backlog/issues/282

#### Changes
1. use `mock-fs` to improve the performance when the test actually writes files to the repo.
2. test `generte()` directly to use `mock-fs` instead of executing the cli with `execa` because `execa` and `mock-fs` are running on different processes hence these cannot use on the same process.
3. move `createOptionsFromConfig()` into `generator-options.ts` in order to test the function directly.

### CODE REVIEW

File: test-packages/e2e-tests/test/generator-cli.spec.ts [hunk 1]
```diff
@@ -30,53 +41,125 @@ describe('generator-cli', () => {
...
-
+  });
+  it('should create options from a config file', () => {
+    mock({ [pathToConfig]: mock.load(pathToConfig) });
+    expect(createOptionsFromConfig(pathToConfig)).toEqual({
+      inputDir,
+      outputDir
+    });
+  });
+  it('should ', () => {
```
Comment (FrankEssenberger) [inline test-packages/e2e-tests/test/generator-cli.spec.ts:73]: [q] old test?

File: test-packages/e2e-tests/test/generator-cli.spec.ts [hunk 2]
```diff
@@ -30,53 +41,125 @@ describe('generator-cli', () => {
...
+        .toString()
     );
     expect(actualPackageJson.version).toEqual('42.23');
-
-    expect(process.stdout).toMatch(
-      /The option 'versionInPackageJson' is deprecated since v2.6.0./
-    );
-  }, 60000);
+  });
+  it('should throw an error for a fail of generation due to an invalid specification file', async () => {
```
Comment (FrankEssenberger) [inline test-packages/e2e-tests/test/generator-cli.spec.ts:146]: [q] I think we wanted to use this early failing test to ensure the warning is still shown.

File: test-packages/e2e-tests/test/generator-cli.spec.ts [hunk 3]
```diff
@@ -30,53 +41,119 @@ describe('generator-cli', () => {
...
+        .toString()
     );
     expect(actualPackageJson.version).toEqual('42.23');
-
-    expect(process.stdout).toMatch(
-      /The option 'versionInPackageJson' is deprecated since v2.6.0./
-    );
-  }, 60000);
+  });
+  it('should throw a warning messge for a deprecated option even when the generatoin process is failed', async () => {
```
Comment (fwilhe) [inline test-packages/e2e-tests/test/generator-cli.spec.ts:139]:
```suggestion
  it('should throw a warning message for a deprecated option even when the generation process is failed', async () => {
```

File: test-packages/e2e-tests/test/generator-cli.spec.ts [hunk 4]
```diff
@@ -30,53 +41,119 @@ describe('generator-cli', () => {
...
+        .toString()
     );
     expect(actualPackageJson.version).toEqual('42.23');
-
-    expect(process.stdout).toMatch(
-      /The option 'versionInPackageJson' is deprecated since v2.6.0./
-    );
-  }, 60000);
+  });
+  it('should throw a warning message for a deprecated option even when the generation process is failed', async () => {
```
Comment (deekshas8) [inline test-packages/e2e-tests/test/generator-cli.spec.ts:139]: [req] This test times out consistently for me. Maybe we can increase the timeout here ?

File: test-packages/e2e-tests/test/generator-cli.spec.ts [hunk 5]
```diff
@@ -30,53 +41,119 @@ describe('generator-cli', () => {
...
+        .toString()
     );
     expect(actualPackageJson.version).toEqual('42.23');
-
-    expect(process.stdout).toMatch(
-      /The option 'versionInPackageJson' is deprecated since v2.6.0./
-    );
+  });
+  it('should throw a warning message for a deprecated option even when the generation process is failed', async () => {
+    try {
```
Comment (FrankEssenberger) [inline test-packages/e2e-tests/test/generator-cli.spec.ts:140]: [pp] Perhaps add a comment why we use the broken service here: "User broken service to stop service generation early - we are only interested in the log statement"

**Review by FrankEssenberger — APPROVED** [review]
LGTM

**Pull request merged**